Mathlib/Algebra/Group/Classes.lean

```diff
@@ -0,0 +1,29 @@
+/-
+Copyright (c) 2014 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Leonardo de Moura, Simon Hudon, Mario Carneiro
+-/
+import Mathlib.Algebra.Group.Defs
+import Mathlib.Init.Algebra.Classes
+
+/-!
+# Instances for operations in algebraic structures.
+
+As the unbundled algebraic structures typeclasses are of somewhat dubious use at present,
+we separate out the file providing instances from the main `Mathlib.Algebra.Group.Defs` file.
```

Comment:
I think `IsCommutative` and `IsAssociative` are part of the few ones we'll keep. See the module doc of `Init.Algebra.Classes`.

Reply:
The module doc says that they are still used, not that they are a good idea, I think.

Reply:
I guess @digama0 can clarify the intended meaning, but we'll need some version of them regardless. Maybe a non-class predicate. But we need *something*.

Reply:
Sure, but I think this is orthogonal to this PR, we don't need any version of them here.